sokutei.h

```diff
@@ -0,0 +1,188 @@
+#ifndef SOKUTEI_BENCHMARK_H
+#define SOKUTEI_BENCHMARK_H
+
+#define SOKUTEI_ITERATIONS 10
+#define SOKUTEI_MAX_COUNTER_COUNT  100
+#define SOKUTEI_COUNTER_LENGTH 30
+#define SOKUTEI_INTEGER_COUNTER_TYPE int
+#define SOKUTEI_FLOAT_COUNTER_TYPE double
+#define SOKUTEI_INTERVAL_TIMER_COUNTER_TYPE int
+
+#define MAX_SIZE_OF_TYPES   sizeof(SOKUTEI_INTEGER_COUNTER_TYPE) > sizeof(SOKUTEI_FLOAT_COUNTER_TYPE) \
+                            ? sizeof(SOKUTEI_INTEGER_COUNTER_TYPE) > sizeof(SOKUTEI_INTERVAL_TIMER_COUNTER_TYPE) \
+                                ? sizeof(SOKUTEI_INTEGER_COUNTER_TYPE) \
+                                : sizeof(SOKUTEI_INTERVAL_TIMER_COUNTER_TYPE)\
+                            : sizeof(SOKUTEI_FLOAT_COUNTER_TYPE) > sizeof(SOKUTEI_INTERVAL_TIMER_COUNTER_TYPE) \
+                                ? sizeof(SOKUTEI_FLOAT_COUNTER_TYPE) \
+                                : sizeof(SOKUTEI_INTERVAL_TIMER_COUNTER_TYPE)
+
+#define SOKUTEI_INTEGER_TYPE '1'
+#define SOKUTEI_FLOAT_TYPE '2'
+#define SOKUTEI_INTERVAL_TYPE '3'
+
+#define NOT_FOUND -1
+#define NOT_MATCHING_TYPE -2
+
+
+int sokutei_strcmp(const char *string_a, const char *string_b){
+    int i = 0;
+    while(string_a[i] == string_b[i] && string_a[i]) i++;
+    return string_a[i] - string_b[i];
+}
+#define STRCMP(string_a, string_b) sokutei_strcmp(string_a, string_b)
+
+char *sokutei_strcpy(char *string_a, const char *string_b){
+    int i;
+    for(i = 0; (string_a[i] = string_b[i]) && (i < SOKUTEI_COUNTER_LENGTH); i++);
+    string_a[SOKUTEI_COUNTER_LENGTH] = '\0';
+    return string_a;
+}
+#define STRCPY(string_a, string_b) sokutei_strcpy(string_a, string_b)
+#define string_equals(string_a, string_b) (STRCMP(string_a, string_b) == 0)
+
+
+
+
+
+
+char sokutei_counter_names[SOKUTEI_MAX_COUNTER_COUNT][SOKUTEI_COUNTER_LENGTH + 2] = {'\0'};
+char sokutei_counters[SOKUTEI_MAX_COUNTER_COUNT * MAX_SIZE_OF_TYPES] = {0};
+int  sokutei_number_of_counters = 0; //this buddy gets overwritten when positioned at 40. line
+
+
+
+char sokutei_get_type_of(int index){
+    return sokutei_counter_names[index][31];
+}
+
+
+int sokutei_get_index_of_counter(const char *counter_name){
+    int i;
+    printf("number of sokutei counters: %d\n", sokutei_number_of_counters);
```

Comment:
You might want to consider introducing a macro like `SOKUTEI_DEBUG_PRINT(x)` which would be defined as a printf if `DEBUG` is defined and would be empty otherwise.

Of course, empty macros are possible: https://stackoverflow.com/questions/9187628/empty-function-macros